yelp_bytes.py

```diff
@@ -26,22 +37,24 @@ def from_bytes(obj, encoding='internet', errors='strict'):
 
     This function always returns unicode.
     """
-    if isinstance(obj, unicode):
+    mro = type(obj).mro()
+    if unicode in mro:
         return obj
+    elif bytes in mro:
+        return obj.decode(encoding, errors)
+    elif hasattr(obj, '__bytes__'):
+        return obj.__bytes__().decode(encoding, errors)
```

Comment:
honestly i don't think this module even makes much sense on python 3, where it's much harder for bytes to sneak into your application, and you really only need to worry about actual `bytes` rather than arbitrary poorly-written objects.

you're basically reintroducing the bug that the doctests illustrate (the bug which is so thoroughly solved that i had to _disable the doctests_ because the illustration no longer applied) just so tests written to exacerbate that bug continue to do so.